### ISSUE TITLE
Add Jetson export for TensorRT on DLA support

### ISSUE THREAD

**Issue by Burhan-Q** [ultralytics/ultralytics#14999]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Per discussion at https://www.jetson-ai-lab.com/research.html#meeting-schedule on 2024-08-06 would be good to include option to enable TensorRT export targeting DLA. 

### Use case

Only enable when `IS_JETSON == True`

### Additional

- [ ] figure out how to re-use an existing export argument
- [ ] run benchmarks comparing

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by lakshanthad** [comment]
@Burhan-Q 

It seems to utilize the [DLA (Deep Learning Accelerator)](https://developer.nvidia.com/deep-learning-accelerator) on NVIDIA Jetson Orin devices, you need to pass extra arguments during the export process of a TensorRT(.engine) file. So I have done a few experiments using the `trtexec` tool which comes built-in on the Jetson.

1. Export an ONNX model using Ultralytics package
```sh
yolo export model=yolov8n.pt format=onnx
```

2. Pass the ONNX model to `trtexec` tool with one of the DLA cores enabled. Here it can be either `0` or `1` since Jetson Orin has 2 DLA cores.
```sh
/usr/src/tensorrt/bin/trtexec --onnx=yolov8n.onnx --saveEngine=yolov8n.engine --int8 --useDLACore=0
```

3. At the middle of the export process, it will fail because some layers are not supported to run on the DLA.

```sh
[08/20/2024-16:34:08] [I] Finished parsing network model. Parse time: 0.0310004
[08/20/2024-16:34:08] [I] FP32 and INT8 precisions have been specified - more performance might be enabled by additionally specifying --fp16 or --best
[08/20/2024-16:34:08] [E] Error[2]: [network.cpp::operator()::2789] Error Code 2: Internal Error (Assertion allowGPUFallback failed. Layer '/model.22/Reshape' is not supported on DLA but GPU fallback is not enabled.)
[08/20/2024-16:34:09] [E] Error[4]: [network.cpp::validate::2901] Error Code 4: Internal Error (DLA validation failed)
[08/20/2024-16:34:09] [E] Engine could not be created from network
[08/20/2024-16:34:09] [E] Building engine failed
[08/20/2024-16:34:09] [E] Failed to create engine from model or file.
[08/20/2024-16:34:09] [E] Engine set up failed
```

4. So we need to add another argument which will allow the unsupported layers to fallback to GPU as follows
```sh
/usr/src/tensorrt/bin/trtexec --onnx=yolov8n.onnx --saveEngine=yolov8n.engine --int8 --useDLACore=0 --allowGPUFallback
```

5. The output will be as follows (the below is only part of the log. The complete log has a lot of layers falling back to GPU):
```sh
[08/20/2024-17:24:53] [W] [TRT] Layer '/model.22/Constant_12_output_0' (CONSTANT): Unsupported on DLA. Switching this layer's device type to GPU.
[08/20/2024-17:24:53] [W] [TRT] Layer '(Unnamed Layer* 299) [Shuffle]' (SHUFFLE): Unsupported on DLA. Switching this layer's device type to GPU.
[08/20/2024-17:24:53] [W] [TRT] /model.22/Concat_5: DLA only supports concatenation on the C dimension.
[08/20/2024-17:24:53] [W] [TRT] Layer '/model.22/Concat_5' (CONCATENATION): Unsupported on DLA. Switching this layer's device type to GPU.
[08/20/2024-17:24:53] [W] [TRT] Calibrator is not being used. Users must provide dynamic range for all tensors that are not Int32 or Bool.
[08/20/2024-17:24:53] [W] [TRT] Running Softmax in INT8 mode on DLA may cause accuracy degradation, so layer: '/model.22/dfl/Softmax' will fall back to GPU. You can force the layer to run on DLA in INT8 mode by setting the layer precision explicitly. Alternatively, you can enable FP16 mode in the optimizer to allow the layer to run on DLA in FP16 mode.
[08/20/2024-17:24:53] [W] [TRT] Dimension: 3 (8400) exceeds maximum allowed size for DLA: 8192
[08/20/2024-17:24:53] [W] [TRT] Validation failed for DLA layer: /model.22/dfl/conv/Conv. Switching to GPU fallback.
[08/20/2024-17:24:53] [W] [TRT] Input tensor has less than 4 dimensions for /model.22/Sub. At least one shuffle layer will be inserted which cannot run on DLA.
[08/20/2024-17:24:53] [W] [TRT] Dimension: 3 (8400) exceeds maximum allowed size for DLA: 8192
[08/20/2024-17:24:53] [W] [TRT] Validation failed for DLA layer: /model.22/Sub. Switching to GPU fallback.
[08/20/2024-17:24:53] [W] [TRT] Input tensor has less than 4 dimensions for /model.22/Add_1. At least one shuffle layer will be inserted which cannot run on DLA.

```

6. The above will successfully export a `.engine` file and this can be used with YOLO validation command
```sh
yolo val model=yolov8n.engine data=coco8.yaml imgsz=640
``` 

However, the problem here is the inference time that we get out of this DLA-enabled model is higher than the inference time we get from a GPU-only-enabled model. This must be because at the inference stage, since DLA and GPU are both used, some layers falling back to GPU costs additional inference time. From a raw inference performance perspective, it seems it is not ideal to implement the DLA functionality to official Ultralytics TensorRT engine export pipelines. 

In conclusion, we can close this issue for now and reopen again if necessary.
